### ISSUE TITLE
Wrong ordering of stylesheets: W3C TR style sheet must be last

### ISSUE THREAD

**Issue by tidoust** [tabatkins/bikeshed#2242]
The latest version of the Pubrules checker (v6.1.4), released last week, seems to check the [W3C TR style sheet must be last](https://www.w3.org/pubrules/doc/rules?profile=WD#lastStylesheet) rule more thoroughly, and now chokes on `<style>` tags that Bikeshed puts after the link to the W3C TR style sheet (both those added by default and custom styles that authors may choose to add).

One example that passed PubRules last week but no longer does is https://www.w3.org/pubrules/?url=https://www.w3.org/TR/2022/WD-webgpu-20220303/&profile=auto

**Comment by jennyliang220** [comment]
The rule existed in Pubrules for a long time, but the check was loose before. It's fixed now and would report an error to documents with custom styles.

To fix this issue, please insert the custom style **before** the W3C stylesheet, so that the W3C stylesheet doesn't get overlayed. @tabatkins 

https://github.com/tabatkins/bikeshed/blob/bda455a127686b75ef54c203463b06a201c7edc4/bikeshed/Spec.py#L287-L289

Good:
```
<style> #id {custom: styles; } </style>
<link rel="stylesheet" href="https://www.w3.org/StyleSheets/TR/2021/W3C-NOTE">
```

**Comment by tidoust** [comment]
@jennyliang220, the code you quoted runs if [Prepare for TR](https://tabatkins.github.io/bikeshed/#metadata-prepare-for-tr) is set. Could it be that Echidna no longer sets that flag for some reason? (or never set it but that went unnoticed?) If I add "Prepare for TR: yes" to the metadata locally, that seems to fix the issue in particular. I don't think that we should require all spec authors to add that directive themselves.

That said, having the same ordering of stylesheets in Editor's Drafts and in TR documents could be a good thing to have, as this may create hard to spot cascading implications (an editor who defines custom styles in an ED would expect these styles to run equally well in the TR version).

**Comment by tidoust** [comment]
Looking at it again, Echidna is not where the problem lies. It's rather the spec-prod action that should set the setting (similar to the way the [`bikeshed echidna`](https://tabatkins.github.io/bikeshed/#cli-echidna) command sets it).

I created an issue against spec-prod (see https://github.com/w3c/spec-prod/issues/128) and am closing this issue as Bikeshed works as intended.

**Comment by tabatkins** [comment]
Yes, echidna should `Prepare TR`, as that's what triggers the W3C stylesheets to move to the end of the `head`.

That said, I was also promised that changes to PubRules like this would be announced ahead of time so that sudden "nobody can publish with Bikeshed" issues like this wouldn't happen again, after the Process 2021 debacle. @plehegar ?

**Comment by svgeesus** [comment]
>  I was also promised that changes to PubRules like this would be announced ahead of time

Yes. It is becoming an increasing problem that a spec which was published a year ago, a month ago, a week ago will now no longer publish because un-discussed and un-announced changes to pubrules are only discovered by the editors or staff contacts when a publication fails.
